scripts/generatePackageSwift.swift

```diff
@@ -5,33 +5,114 @@
 
-import Foundation
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+import class Foundation.PropertyListDecoder
+import class Foundation.ProcessInfo
+import class Foundation.FileManager
 
+let env = ProcessInfo.processInfo.environment
+
+// This struct is read from the .plist stored at versionDependencies.plist
 struct VersionDeps: Codable {
+    // Versions will always be defined in versionDependencies.plist
+    // Versions are the only reference used when releasing.
     var awsCRTSwiftVersion: String
     var clientRuntimeVersion: String
+    // Branches are not normally defined in versionDependencies.plist,
+    // but may be set during development on a feature branch if desired.
+    // These values override environment vars set on CI.
+    // Branches are ignored when building a release.
+    var awsCRTSwiftBranch: String?
+    var clientRuntimeBranch: String?
+    // Paths may be used to point to paths on a development machine.
+    // They may be set by the developer during testing, but should
+    // never be set outside a development branch.
+    // Use of paths will cause CI to fail.
+    // Paths are ignored when building a release.
+    var awsCRTSwiftPath: String?
+    var clientRuntimePath: String?
+}
+
+let includeProtocolTests: Bool = {
+    env["AWS_SDK_PROTOCOL_CODEGEN_TESTS_BUILT"] != nil
+}()
+
+let releaseInProgress: Bool = {
+    env["AWS_SDK_RELEASE_IN_PROGRESS"] != nil
+}()
+
+struct ProtocolTest {
+    let name: String
+    let path: String
+    let local: Bool
 }
+
+let protocolBasePath = "./codegen/protocol-test-codegen/build/smithyprojections/protocol-test-codegen"
+let protocolBasePathLocal = "./codegen/protocol-test-codegen-local/build/smithyprojections/protocol-test-codegen-local"
+
+let protocolTests: [ProtocolTest] = {
+    guard includeProtocolTests else { return [] }
+    return [
+        ProtocolTest(name: "AWSRestJsonTestSDK", path: "aws-restjson", local: false),
+        ProtocolTest(name: "AWSJson1_0TestSDK", path: "aws-json-10", local: false),
+        ProtocolTest(name: "AWSJson1_1TestSDK", path: "aws-json-11", local: false),
+        ProtocolTest(name: "RestXmlTestSDK", path: "rest-xml", local: false),
+        ProtocolTest(name: "RestXmlWithNamespaceTestSDK", path: "rest-xml-xmlns", local: false),
+        ProtocolTest(name: "Ec2QueryTestSDK", path: "ec2-query", local: false),
+        ProtocolTest(name: "AWSQueryTestSDK", path: "aws-query", local: false),
+        //Service specific
+        ProtocolTest(name: "APIGatewayTestSDK", path: "apigateway", local: false),
+        ProtocolTest(name: "GlacierTestSDK", path: "glacier", local: false),
+        ProtocolTest(name: "MachineLearningTestSDK", path: "machinelearning", local: false),
+        ProtocolTest(name: "S3TestSDK", path: "s3", local: false),
+        //Local tests
+        ProtocolTest(name: "aws_restjson", path: "aws-restjson", local: true),
+        ProtocolTest(name: "rest_json_extras", path: "rest_json_extras", local: true),
```

Comment:
we must not hardcode these because they are dynamically generated and more can be added, this must done through reading the generated test packages

Reply:
They were hardcoded in the original `codegen/Package.swift`.  Was that file itself auto generated from somewhere?